helper/pluginutil/runner.go

```diff
@@ -98,6 +102,39 @@ func (r *PluginRunner) Run(wrapper RunnerUtil, pluginMap map[string]plugin.Plugi
 	return client, nil
 }
 
+func (r *PluginRunner) RunMetadataMode(wrapper RunnerUtil, pluginMap map[string]plugin.Plugin, hs plugin.HandshakeConfig, env []string, logger log.Logger) (*plugin.Client, error) {
+	cmd := exec.Command(r.Command, r.Args...)
```

Comment:
This appears to be, other than whether metadatamode env var is enabled or not, the same exact logic as the latter part of Run. Any chance they can be combined into a runCommon that Run and RunMetadataMode call with appropriate bool var?

Reply:
@briankassouf reverted this on https://github.com/hashicorp/vault/commit/5526108c3df336281cb2f52a550c5f526459dcf6#diff-217b9ed237f303ae16a69462dfcc6bf3. I think it's because he wanted to keep the behavior separate explicitly. Brian correct me if otherwise :)

Reply:
That's correct, but i'm okay with `RunMetadataMode`, `Run`, and `runCommon`. Other plugin types shouldn't have to know about metadata mode IMO.